pypdf/filters.py

```diff
@@ -564,6 +564,8 @@ def _get_parameters(
         k = 0
         columns = 1728
         if parameters:
+            if isinstance(parameters, IndirectObject):
```

Comment:
the test is not mandatory: you should be able to just have 
`parameters = parameters.get_object()` 

Also you should add `IndirectObject` in the list of allowed types in line 561

Reply:
Thanks for the hint. I have updated the PR.